Add recipe for dicom

minad opened the pull request:
### Brief summary of what the package does

DICOM stands for Digital Imaging and Communications in Medicine. DICOM files are typically used for medical imaging with different modalities like US, CR, CT, MRI or PET. This package adds the ability to view such files in Emacs. The images and metadata are displayed in regular Emacs buffers. The package registers itself in auto-mode-alist and magic-mode-alist for DICOMDIR directory files and DICOM images (file extension *.dcm or *.ima). Furthermore the command dicom-open opens DICOMDIR directory files or DICOM image files interactively.

### Direct link to the package repository

https://github.com/minad/dicom

### Your association with the package

Author and maintainer

### Relevant communications with the upstream package maintainer

**None needed**

### Checklist

<!-- Please confirm by replacing `[]` with `[x]`: -->

- [X] The package is released under a [GPL-Compatible Free Software License](https://www.gnu.org/licenses/license-list.en.html#GPLCompatibleLicenses)
- [X] I've read [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)
- [X] I've used the latest version of [package-lint](https://github.com/purcell/package-lint) to check for packaging issues, and addressed its feedback
- [X] My elisp byte-compiles cleanly
- [X] I've used `M-x checkdoc` to check the package's documentation strings
- [X] I've built and installed the package using the instructions in [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)

<!-- After submitting, please fix any problems the CI reports. -->

riscy commented:
Thanks for this.  I see this is already on ELPA; here's a quick first pass -

⸺ `dicom.el` with [melpazoid](https://github.com/riscy/melpazoid):
```
- dicom.el#L625: `(funcall 'eval-after-load 'ol` - this seems convoluted?  Alterntively, if this is necessary, this is a good spot to use a sharp-quote
- dicom.el#L635: You may want to consider `(derived-mode-p 'xyz)`
```

⸺ Package and license:
- Please specify `:fetcher` before `:repo` in your recipe
- Reminder: if you haven't, ensure GitHub release 0.4 is up-to-date with your current code and `Package-Version`
- pkg/dicom.el: GPL*

minad commented:
> Thanks for this. I see this is already on ELPA; here's a quick first pass -

Yes, all my packages are on ELPA and MELPA since I prefer to make them avaiable via both channels. I've assigned copyright to the FSF.

> - dicom.el#L625: `(funcall 'eval-after-load 'ol` - this seems convoluted?  Alterntively, if this is necessary, this is a good spot to use a sharp-quote

Fixed. A sharp-quote cannot be used since `dicom-mode` is not autoloaded. This will lead to a warning in the autoload file.

> - dicom.el#L635: You may want to consider `(derived-mode-p 'xyz)`

Fixed.

> Please specify :fetcher before :repo in your recipe

Fixed.

> Reminder: if you haven't, ensure GitHub release 0.4 is up-to-date with your current code and Package-Version

The Version: keyword bump is consistent with the 0.4 version tag on Github.

riscy commented:
> A sharp-quote cannot be used since dicom-mode is not autoloaded

I think before you had `funcall 'eval-after-load` - I should've specified the function `eval-after-load` was the one I was thinking of sharp-quoting, but I see you removed the `funcall` in the latest.  The change of course reveals the package-lint warning - `625:1: warning: 'eval-after-load' is for use in configurations, and should rarely be used in packages.` but this looks appropriate in this case.

Looks good - thanks again!

minad commented:
> The change of course reveals the package-lint warning - 625:1: warning: 'eval-after-load' is for use in configurations, and should rarely be used in packages. but this looks appropriate in this case.

Yes, the funcall was only there to prevent the warning. ;)

It is ugly, but I do not see another way of registering the Org links. The autoload is needed to make links work before loading dicom.el. And the eval-after-load is needed to delay registering the links until Org is loaded.
